q2_types/feature_data/_format.py

```diff
@@ -90,38 +90,59 @@ class TSVTaxonomyFormat(model.TextFileFormat):
 
     Optionally followed by other arbitrary columns.
 
-    This format supports comment lines starting with #, and blank lines. The
-    expected header must be the first non-comment, non-blank line. In addition
-    to the header, there must be at least one line of data.
+    This format supports blank lines. The expected header must be the first
+    non-blank line. In addition to the header, there must be at least one line
+    of data.
 
     """
     HEADER = ['Feature ID', 'Taxon']
 
-    def sniff(self):
+    def _check_n_records(self, n=None):
         with self.open() as fh:
             data_lines = 0
             header = None
-            while data_lines < 10:
-                line = fh.readline()
 
+            file_ = enumerate(fh) if n is None else zip(range(n), fh)
+
+            for i, line in file_:
+                # Tracks line number for error reporting
+                i = i + 1
                 if line == '':
```

Comment:
Based on looking at coverage, it looks like we don't currently have any tests with files that contain blank lines, since this file format is supposed to support blank lines maybe we want to add one? What do you think @thermokarst

Reply:
> it looks like we don't currently have any tests with files that contain blank lines

Allow me to introduce you to this beauty: https://github.com/qiime2/q2-types/blob/master/q2_types/feature_data/tests/data/taxonomy/valid-but-messy.tsv

I think you might be misreading this code branch --- the line you have commented on is for the EOF, the next branch handles blank lines.

It does kinda look to me like this EOF branch isn't actually doing anything though, eh?

Reply:
Yeah, I must've gone cross-eyed when reading that or something :laughing:. And no, it doesn't seem to do anything because presumably Python doesn't give us empty lines when it iterates over files

Reply:
Thanks for the feedback @thermokarst and @Oddant1,

The `line == ''` condition will not be hit due to how Python for loops treat lines in a file. The use of the `zip` function (in the `file_` variable declaration) can be a maximum size of range `n` or the length of `fh`. This value is determined before the start of the loop (because of the zip function call) and is used by the for loop to iterate through `file_`. No line will truly be empty because a new line character or taxonomy data will be present at each line.

I have already pushed a new commit to remove the condition from `_format.py`.